packages/react-components/react-table/src/hooks/useColumnResizeMouseHandler.ts

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { TableColumnId, ColumnResizeState } from './types';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+export default function useColumnResizeMouseHandler(columnResizeState: ColumnResizeState) {
+  const mouseX = React.useRef(0);
+  const currentWidth = React.useRef(0);
+  const colId = React.useRef<TableColumnId | undefined>(undefined);
+
+  const { targetDocument } = useFluent();
+  const globalWin = targetDocument?.defaultView;
+
+  const handleMouseMove = React.useCallback(
+    (e: MouseEvent) => {
+      const dx = e.clientX - mouseX.current;
+
+      // Update the local width for the column and set it
+      currentWidth.current += dx;
+      colId.current && columnResizeState.setColumnWidth(colId.current, currentWidth.current);
+      mouseX.current = e.clientX;
+    },
+    [columnResizeState],
+  );
+
+  const onMouseMove = React.useCallback(
```

Comment:
nit: `useEventCallback` can be used here

Reply:
nice, trying it out, thx